src/bootstrap/src/lib.rs

```diff
@@ -84,38 +84,9 @@ const EXTRA_CHECK_CFGS: &[(Option<Mode>, &str, Option<&[&'static str]>)] = &[
     (Some(Mode::ToolRustc), "rust_analyzer", None),
     (Some(Mode::ToolStd), "rust_analyzer", None),
     (Some(Mode::Codegen), "parallel_compiler", None),
-    // NOTE: consider updating `check-cfg` entries in `std/Cargo.toml` too.
-    // cfg(bootstrap) remove these once the bootstrap compiler supports
-    // `lints.rust.unexpected_cfgs.check-cfg`
-    (Some(Mode::Std), "stdarch_intel_sde", None),
```

Comment:
Why is it ok to just remove these, instead of adding the to `lints.rust.unexpected_cfgs.check-cfg` (once the bootstrap compiler can understand that?)?

Reply:
because the bootstrap compiler (as of Rust 1.80) can now understand `lints.rust.unexpected_cfgs.check-cfg`

and also because it is already in the table, cf. https://github.com/rust-lang/rust/blob/a4ce33c0b232deda1cbce447e80f187cd34952a6/library/core/Cargo.toml#L44-L47

Reply:
to give a bit of context, we added in #125452 some cfgs directly in the `Cargo.toml` for `-Zbuild-std` as it uses the nightly compiler which understood and applied the `check-cfg` lint config; but because the bootstrap compiler used previous beta we couldn't removed them (yet) from `bootstrap`

this PR does that cleanup by having the source of truth for the standard library crates directly inside their `Cargo.toml` instead of in bootstrap